File: mycollab-web/src/main/java/com/mycollab/vaadin/reporting/FormReportStreamSource.java
package com.mycollab.vaadin.reporting;

import com.mycollab.core.MyCollabException;

import java.util.Map;

/**
 * @author MyCollab Ltd
 * @since 5.2.11
 */
public class FormReportStreamSource<B> extends ReportStreamSource {
    private B bean;
    private FormReportLayout formReportLayout;

    public FormReportStreamSource(FormReportTemplateExecutor<B> templateExecutor) {
        super(templateExecutor);
    }

    public B getBean() {
        return bean;
    }

    public void setBean(B bean) {
        this.bean = bean;
    }

    public void setFormLayout(FormReportLayout formReportLayout) {
        this.formReportLayout = formReportLayout;
    }

    @Override
    protected void initReportParameters(Map<String, Object> parameters) {
        if (bean != null) {
            parameters.put("bean", bean);
        } else {
            throw new MyCollabException("Bean must be not null");
        }

        parameters.put("layout", formReportLayout);
    }
}

File: mycollab-web/src/main/java/com/mycollab/vaadin/reporting/FormReportStreamSource.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package com.mycollab.vaadin.reporting

import com.mycollab.core.MyCollabException

/**
 * @author MyCollab Ltd
 * @since 5.2.11
 */
class FormReportStreamSource<B>(templateExecutor: FormReportTemplateExecutor<B>) : ReportStreamSource(templateExecutor) {
    var bean: B? = null
    private lateinit var formReportLayout: FormReportLayout

    fun setFormLayout(formReportLayout: FormReportLayout) {
        this.formReportLayout = formReportLayout
    }

    override fun initReportParameters(parameters: MutableMap<String, Any>) {
        if (bean != null) {
            parameters.put("bean", bean!!)
        } else {
            throw MyCollabException("Bean must be not null")
        }

        parameters.put("layout", formReportLayout)
    }
}
